edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js

```diff
@@ -0,0 +1,247 @@
+
+            /* Load the static template for rendering. */
+            this.loadTemplateData();
+        },
+        events: {
+            'click .search-onboarding > span.search': 'searchItems',
+            'click .search-onboarding > span.clear-search': 'clearSearch',
+            'click .status-checkboxes > li > input': 'filter',
+            'click li > a.target-link': 'getPaginatedAttempts'
+        },
+        searchItems: function(event) {
+            var searchText = $('#search_onboarding_id').val();
+            if (searchText !== '') {
+                this.inSearchMode = true;
+                this.searchText = searchText;
+                this.currentPage = 1;
+                this.collection.url = this.constructUrl();
+                this.hydrate();
+                event.stopPropagation();
+                event.preventDefault();
+            }
+        },
+        clearSearch: function(event) {
+            this.inSearchMode = false;
+            this.searchText = '';
+            this.currentPage = 1;
+            this.collection.url = this.constructUrl();
+            this.hydrate();
+            event.stopPropagation();
+            event.preventDefault();
+        },
+        filter: function(event) {
+            var $checkbox = $(event.currentTarget);
+            var checked = $checkbox.is(':checked');
+            var value = $checkbox.val();
+            var index = this.filters.indexOf(value);
+            $checkbox.prop('disabled', true);
+            this.currentPage = 1;
+            if (checked && !this.filters.includes(value)) {
+                this.filters.push(value);
+            } else if (!checked && this.filters.includes(value)) {
+                this.filters.splice(index, 1);
+            } else {
+                // prevent extra calls due to double-clicking
+                return;
+            }
+            this.collection.url = this.constructUrl();
+            this.hydrate();
+            $checkbox.prop('disabled', false);
+            event.stopPropagation();
+            event.preventDefault();
+        },
+        constructUrl: function(page) {
+            var url;
+            page = typeof page !== 'undefined' ? page : null; // eslint-disable-line no-param-reassign
+            // if the page has changed, update the current page
+            if (page) {
+                this.currentPage = page;
+            }
+            url = this.initialUrl + this.courseId + '?page=' + this.currentPage;
+            if (this.searchText) {
+                url = url + '&text_search=' + this.searchText;
+            }
+            if (this.filters.length > 0) {
+                url += '&statuses=';
+                this.filters.forEach(function(filter, i) {
+                    if (i > 0) {
+                        url += ',';
+                    }
+                    url += filter;
+                });
+            }
+            return url;
+        },
+        getPaginatedAttempts: function(event) {
```

Comment:
#NIT I feel like the function can be better named as `getPaginatedData`? Make it sound more generic make it easier to read